File: Corrections/java/Neufplate/src/main/java/fr/neufplate/PriceCalculator/NftPrice.java
package fr.neufplate.PriceCalculator;

import fr.neufplate.Nft;

public class NftPrice extends Price {

    private final int computerPrice = 4000;
    private final int screenTimeAverageByDay = 5;
    private final int computerLifetimeInYears = 3;
    private final double timeElapsedInSecond;

    public NftPrice(long startTime, long endTime) {
        float timeElapsedInMilliSecond = endTime - startTime;
        //System.out.println("timeElapsedInMilliSecond = " + timeElapsedInMilliSecond);

        this.timeElapsedInSecond =  timeElapsedInMilliSecond / 1000;
        //System.out.println("timeElapsedInSecond = " + timeElapsedInSecond);
    }

    @Override
    public double calculate() {
        int computerLifetime = screenTimeAverageByDay * 365 * computerLifetimeInYears * 60;
        double nftPricePerMinute = (double) computerPrice / computerLifetime;
        double nftPricePerSecond = nftPricePerMinute / 60;
        double price = nftPricePerSecond * timeElapsedInSecond;
        this.baseValue = price;
        return price;
    }
}

File: Corrections/php/Neufplate/src/PriceCalculator/NftPrice.php
<?php

namespace App\PriceCalculator;

class NftPrice extends Price
{

    private const computerPrice = 4000;
    private const screenTimeAverage = 5;
    private const computerLifetimeInYears = 3;

    private int $timeElapsedInSecond;

    public function __construct($startTime, $endTime)
    {
        $this->timeElapsedInSecond = $endTime - $startTime;
    }


    public function calculate(): float
    {
        $computerLifetimeInHour = self::computerLifetimeInYears * 365 * self::screenTimeAverage;
        $nftPricePerSecond = self::computerPrice / $computerLifetimeInHour / 60;
        $price = $nftPricePerSecond * $this->timeElapsedInSecond;
        $this->baseValue = $price;
        return $price;
    }
}